### PR TITLE
trivial spell error in user-facing messages

### PR DESCRIPTION
**Pull request by michaeldesu**

### CODE REVIEW

File: book/nullsafe.tex [hunk 1]
```diff
@@ -77,12 +77,12 @@ \section{Smart null checking}
...
 
 
-Furthermore, in cases where a nullable variable is assigned a non nullable value, we know with certainty that said variable is now not nullable, Concurnas is able to make use of this information thus enabling the following code to be valid:
+Furthermore, in cases where a nullable variable is assigned a non-nullable value, we know with certainty that said variable is now not nullable, Concurnas is able to make use of this information thus enabling the following code to be valid:
 
 \begin{lstlisting}
 nullable String? = "not null"
 //... potentially other code...
-ok = nullable.length()//ok as we know at this point nullable isn't nullable
+ok = nullable.length() //ok as we know at this point nullable isn't nullable
```
Comment (michaeldesu) [inline book/nullsafe.tex]: cosmetic space insertion to make it look better for reader

File: book/nullsafe.tex [hunk 2]
```diff
@@ -15,21 +15,21 @@
 Concurnas, like other modern programming languages such as \href{https://kotlinlang.org/}{Kotlin} and \href{https://developer.apple.com/swift/}{Swift}, makes working with \lstinline{null} easy and safe by incorporating nullability a part of the type system, and by providing a range of null safety operators to assist in those instances where null is a desired object state. This allows us to write safe code in Concurnas which is largely free of \lstinline{NullPointerException} exceptions.
 
 \section{Type system nullability}
-All non primitive types in Concurnas are non nullable at the point of declaration by default. Thus, attempting to assign a null value to a variable having a non nullable type (or anywhere else where a non nullable type is expected, for instance in a method argument etc) will result in a compile time error:
+All non-primitive types in Concurnas are non-nullable at the point of declaration by default. Thus, attempting to assign a null value to a variable having a non-nullable type (or anywhere else where a non-nullable type is expected, for instance in a method argument etc) will result in a compile time error:
```
Comment (michaeldesu) [inline book/nullsafe.tex]: these (and others like it) should be hyphenated

File: book/nullsafe.tex [hunk 3]
```diff
@@ -172,98 +172,98 @@ \section{No null assertion}
...
+In the above code where Concurnas attempts to avoid "null pollution" - if an unknown nullability type resolves to a null value and is set to a non-nullable variable (or say passed as an argument to a method expecting a non-null parameter) then this will result in a \lstinline{NullPointerException}. If the \lstinline{alist} variable were to be created with a nullable generic type: \lstinline{alist = new list<String?>()} then this logic would not be required, though of course we'd be unable to assign the return value of \lstinline{alist.get(0)} to the non-nullable variable: \lstinline{res2 String}. Throwing a \lstinline{NullPointerException} in order to avoid "null pollution" is not desirable, but necessary - and is a beneficial approach where the alternative is permitting "null pollution" because at least the \lstinline{NullPointerException} is thrown at the point of assignment/usage.
```
Comment (michaeldesu) [inline book/nullsafe.tex]: 'non null' to 'non-null'

File: book/nullsafe.tex [hunk 4]
```diff
@@ -314,10 +314,10 @@ \section{Where can NullPointerException's still occur}
...
-	\item \textbf{When using non Concurnas code.} Concurnas has no control over code authored in other languages, so they may throw a \lstinline{NullPointerException}.
+	\item \textbf{Values from non-Concurnas code.} Concurnas attempts to avoid "null pollution" - as such it's possible for a \lstinline{NullPointerException} exception to be thrown from checking that the return value of a unknown nullability type is non-null. This "null pollution" avoidance is not applied to the individual values of arrays as it would not be efficient to check every array in this manner. As such non-primitive arrays and n+1 dimensional arrays provided by non-Concurnas code should be used with care and attention to the possibility of null values being present within them.
```
Comment (michaeldesu) [inline book/nullsafe.tex]: 'non Concurnas' to 'non-Concurnas'

File: book/imports.tex [hunk 5]
```diff
@@ -8,7 +8,7 @@
...
+Imports are essentially referential sugar. They help us to avoid the need to type the fully-qualified path of the asset we are making use of, instead being able to refer to the asset by its short name (the text after the full package name final dot) or by a name we assign at import point (via the \lstinline{as} keyword). At compile time any references to the short name we are making use of are mapped to the fully-qualified path before being passed to the classloader (default or otherwise) for loading. For example, a fully-qualified package name for class \lstinline{MyClass} might be: \lstinline{com.mycode.MyClass}, the short name is \lstinline{MyClass}.
```
Comment (michaeldesu) [inline book/imports.tex]: 'fully qualified' -> 'fully-qualified'

File: book/imports.tex [hunk 6]
```diff
@@ -67,10 +67,10 @@ \section{Import sites}
...
 
 \section{Packages}
-An assets fully qualified importable package name is a function of its name within the module it's declared within, the module name and its path relative to the root of compilation at compilation time. Note that Concurnas does not have a package keyword, instead it relies upon the directory structure relative to the root of compilation in order to determine this. So for example, using a conventional directory structure (found in almost all operating systems) when we compile our code if our root was set to \lstinline{/home/project/src} and our code within this root, in a directory structure \lstinline{./com/mycode.conc} (i.e. file \lstinline{mycode.conc} is within subdirectory \lstinline{com} containing the class definition \lstinline{MyClass}) - then the fully qualified package name of the at compilation time would be \lstinline{com.mycode.MyClass}.
```
Comment (michaeldesu) [inline book/imports.tex]: this part 'then the fully qualified package name of the at compilation time would be
com.mycode.MyClass' I think is missing a word (after this part 'name of the') but not sure what it is? I've not updated this sentence.

Comment (jasontatton) [reply book/imports.tex]: It should read "...then the fully qualified package name of the class at compilation time would be..." - I will fix this in a separate commit

**Comment by jasontatton** [comment]
This is awesome, thanks so much!

**Pull request merged**